mailbox/cassandra/src/main/java/org/apache/james/mailbox/cassandra/mail/CassandraMailboxMapper.java

```diff
@@ -166,6 +166,24 @@ private Mailbox addAcl(MailboxACL acl, Mailbox mailbox) {
                 () -> LOGGER.warn("Could not retrieve mailbox {} with path {} in mailbox table.", idAndPath.getCassandraId(), idAndPath.getMailboxPath())));
     }
 
+    @Override
+    public MailboxId create(Mailbox mailbox) throws MailboxException {
+        CassandraId cassandraId = retrieveId(mailbox);
```

Comment:
these two lines looks horrible. If you don't look at the implementation of retrieveId, it looks like we get the CassandraId from mailbox and then set it back into mailbox.

Reply:
Actually your are right it shouldn't be needed anymore. I should as well change the behavior of `save` to `rename` (it shouldn't be able to create too), and adapt other test classes using it accordingly. Will work on this !